src/dataset/impl_dataset.rs

```diff
@@ -120,6 +120,16 @@ impl<R: Records, S> DatasetBase<R, S> {
     }
 }
 
+impl<X, Y> Dataset<X, Y> {
+    // Converts Ix2 targets to Ix1. Only works for targets with shape of form [X, 1]
```

Comment:
`Convert 2D targets to 1D. Only works for targets with shape of form [X, 1], panics otherwise.`